libraries/AC_Avoidance/AC_Avoid.cpp

```diff
@@ -216,6 +235,47 @@ void AC_Avoid::adjust_velocity(Vector3f &desired_vel_cms, bool &backing_up, floa
             }
         }
     }
+    // limit acceleration
+    limit_accel(desired_vel_cms_original, desired_vel_cms, dt);
+}
+
+// Limit acceleration so that change of velocity output by avoidance library is controlled
+void AC_Avoid::limit_accel(const Vector3f &original_vel, Vector3f &modified_vel, float dt)
+{
+    if (original_vel == modified_vel || is_zero(_accel_max) || is_zero(dt)) {
+        // we can't limit accel if any of these conditions are true
+        return;
+    }
+    // max accel in cm
+    const float max_accel_cm = _accel_max * 100.0f;
+    // accel desired by avoidance library
+    const Vector3f accel = (modified_vel - original_vel)/dt;
+    // max velocity allowed in this time period
+    const float max_vel_diff = max_accel_cm * dt;
+
+    // check if acceleration in any axis is greater than the maximum allowed
```

Comment:
this method of limiting per axis means that the result is different depending upon the direction.  It would probably be better to use the length I think and then scale back both x and y if the length is greater than the maximum.